pytorch_lightning/plugins/precision/sharded_native_amp.py

```diff
@@ -31,7 +31,12 @@ class ShardedNativeMixedPrecisionPlugin(NativeMixedPrecisionPlugin):
     def __init__(self) -> None:
         super().__init__()
         self.scaler = ShardedGradScaler()
+        self.clip_grad_funcs = {  # declaration for adapt overrided function.
+            GradClipAlgorithmType.VALUE: super().clip_grad_by_value,
+            GradClipAlgorithmType.NORM: self.clip_grad_by_norm,
+        }
```

Comment:
- here you could update `clip_grad_funcs` with the new definition for `clip_grad_by_norm` only, since after L32, the property is set. this way in case there are new grad clip algorithm types added in the base, we'll automatically get them here
- on the other hand, if we think that precision plugins will be so specialized that we should explicitly opt-in to different grad clip algorithm types, then this set up is fine

Reply:
Thanks for your comment! Let's erase it. We can use `redeclaration` after we need it later.